src/reactions/ReactionUserList.js

```diff
@@ -25,19 +24,15 @@ class ReactionUserList extends PureComponent<Props> {
   };
 
   render() {
-    const { reactedUserIds, allUsersById } = this.props;
+    const { reactedUserIds } = this.props;
 
     return (
       <FlatList
         data={reactedUserIds}
         keyExtractor={userId => `${userId}`}
-        renderItem={({ item }) => {
-          const user = allUsersById.get(item);
-          if (!user) {
-            return null;
-          }
-          return <UserItem key={user.user_id} user={user} onPress={this.handlePress} />;
-        }}
+        renderItem={({ item }) => (
+          <UserItemById key={item} user={item} onPress={this.handlePress} />
```

Comment:
17bf4a53fe546ea3e9fcaadfb7f1a9ca76f312a7  user: Use UserItemById in ReactionUserList; simplify whole call stack.

Hmm, I'm seeing that `UserItemById`'s props aren't getting type-checked; do you see that too? I notice that the `user` prop given here is a `number`, and that the `userId` prop isn't given. I also notice that the callsite in `GroupDetailsScreen` does pass a `number` for `userId` (that sounds right to me), and the callsite in `PmConversationList` gives a `number` for `user` and doesn't give `userId`.

I believe the intended interface is that callers pass `userId` (a `number`) and don't pass `user` because that'll be supplied (as a `UserOrBot`) by `connect`.

I also find that I can give, e.g.,`foo="nonsense"` in one of `UserItemById`'s callsites without getting Flow errors.

Reply:
Hmm! Thanks for catching this. Will investigate.

Reply:
We discussed this [in chat](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Flow.20and.20.60connect.60/near/1098203) and found a satisfying workaround. It seems to be a Flow bug -- likely the same underlying Flow bug that was behind #3768.